XenAdmin/Commands/MoveVMCommand.cs

```diff
@@ -66,16 +66,22 @@ public MoveVMCommand(IMainWindow mainWindow, VM vm)
 
         protected override void ExecuteCore(SelectedItemCollection selection)
         {
-            VM vm = (VM)selection[0].XenObject;
 
-            if (CrossPoolMoveVMCommand.CanExecute(vm, null))
+            if (selection.AllItemsAre<VM>() && new CrossPoolMoveVMCommand(MainWindowCommandInterface, selection).CanExecute())

```

Comment:
Yep you are right, it's not necessary there. It makes sense to check if selection contains more than one VM to use that execution path